ArenaEnemyCooldown.lua

```diff
@@ -0,0 +1,239 @@
+        -- Check reset by power
+        for i = 1, #resetByPower do
+            local reset = resetByPower[i];
+            if self.activeMap[reset] then
+                local cost = GetSpellPowerCost(spellId);
+                if cost and cost[1] and ( cost[1].type == spellData[reset].reduce_power_type ) then
+                    local amount = spellData[reset].reduce_amount * cost[1].cost;
+                    NS.ResetWeakAuraCooldown(self.activeMap[reset], amount);
+                end
+            end
+        end
+
+        -- Check regular resets
+        if spellResets[spellId] then
+            for resetSpellID, amount in pairs(spellResets[spellId]) do
+                if self.activeMap[resetSpellID] then
+                    NS.ResetWeakAuraCooldown(self.activeMap[resetSpellID], amount);
+                end
+            end
+        end
+    end
+
+    -- Check resets by crit damage (e.g., combustion)
+    if ( subEvent == "SPELL_DAMAGE" ) and critical and ( sourceGUID == guid ) then
+        for i = 1, #resetByCrit do
+            local reset = resetByCrit[i];
+            if self.activeMap[reset] then
+                local spells = spellData[reset].critResets;
+                for i = 1, #spells do
+                    if ( spellId == spells[i] ) or ( spellName == spells[i] ) then
+                        NS.ResetWeakAuraCooldown(self.activeMap[reset], spellData[reset].critResetAmount);
+                    end
+                end
+                return;
+            end
+        end
+    end
+
+    -- Check summon / dead
+    if ( subEvent == "UNIT_DIED" ) then
+        -- Might have already been dismissed by SPELL_AURA_REMOVED, e.g., Psyfiend
+        local summonSpellId = self.npcMap[destGUID];
+        if summonSpellId and self.activeMap[summonSpellId] then
+            NS.ResetWeakAuraDuration(self.activeMap[summonSpellId]);
+        end
+        return;
+    elseif ( subEvent == "SPELL_SUMMON" ) and ( guid == sourceGUID ) then
+        -- We don't actually show the icon from SPELL_SUMMON, just create the mapping of mob GUID -> spellID
+        local npcId = NS.GetNpcIdFromGuid(destGUID);
+        local summonSpellId = npcToSpellID[npcId];
+        self.npcMap[destGUID] = summonSpellId;
+
+        -- If not added yet, add by this (e.g., Guldan's Ambition: Pit Lord)
+        if summonSpellId and self.icons[summonSpellId] and ( not self.activeMap[summonSpellId] ) then
+            NS.StartWeakAuraIcon(self.icons[summonSpellId]);
+        end
+        return;
+    end
+        
+    -- Validate spell
+    if ( not spellData[spellId] ) then return end
+    local spell = spellData[spellId];
+
+    -- Validate unit
+    local spellGUID = ( spell.trackDest and destGUID ) or sourceGUID;
+    if ( spellGUID ~= guid ) then return end
+
+    -- Check spell dismiss
+    if ( subEvent == "SPELL_AURA_REMOVED" ) then
+        if self.activeMap[spellId] then
+            NS.ResetWeakAuraDuration(self.activeMap[spellId]);
```

Comment:
Skyfury is dismissed prematurely

Reply:
Fixed by adding 15s duration, if dismissed early, will be dealt by spell_aura_removed